indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -422,6 +423,31 @@ def _process(self):
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
 
 
+class RHBookingOccurrenceLinkActions(RHBookingBase):
+    """Links a Reservation occurrence to an event."""
+
+    def _process_args(self):
+        RHBookingBase._process_args(self)
+        date = dateutil.parser.parse(request.view_args['date'], yearfirst=True).date()
+        self.occurrence = self.booking.occurrences.filter(ReservationOccurrence.date == date).one()
```

Comment:
No strong opinion but I'd go for the shorter `filter_by` version here:

```suggestion
        self.occurrence = self.booking.occurrences.filter_by(date=date).one()
```

Reply:
Fixed.